src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1123,6 +1123,10 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
                     scriptConfig.Functions.Add((string)function);
                 }
             }
+            else
+            {
+                scriptConfig.Functions = null;
```

Comment:
See RunAndBlock: https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Host/ScriptHostManager.cs#L112 -- we don't create a new ScriptHostConfig. It's the same one for every host.